leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java

```diff
@@ -219,17 +220,17 @@ public void json_encode_device_object_instance() {
     @Test
     public void json_encode_timestamped_resources() throws CodecException {
         List<TimestampedLwM2mNode> data = new ArrayList<>();
-        data.add(new TimestampedLwM2mNode(500L, LwM2mSingleResource.newFloatResource(1, 22.9)));
-        data.add(new TimestampedLwM2mNode(510L, LwM2mSingleResource.newFloatResource(1, 22.4)));
-        data.add(new TimestampedLwM2mNode(520L, LwM2mSingleResource.newFloatResource(1, 24.1)));
+        data.add(new TimestampedLwM2mNode(Instant.ofEpochSecond(500), LwM2mSingleResource.newFloatResource(1, 22.9)));
+        data.add(new TimestampedLwM2mNode(Instant.ofEpochSecond(510), LwM2mSingleResource.newFloatResource(1, 22.4)));
+        data.add(new TimestampedLwM2mNode(Instant.ofEpochSecond(520), LwM2mSingleResource.newFloatResource(1, 24.1)));
 
         byte[] encoded = encoder.encodeTimestampedData(data, ContentFormat.JSON, new LwM2mPath(1024, 0, 1), model);
 
         StringBuilder b = new StringBuilder();
         b.append("{\"bn\":\"/1024/0/1\",\"e\":[");
-        b.append("{\"v\":22.9,\"t\":500},");
-        b.append("{\"v\":22.4,\"t\":510},");
-        b.append("{\"v\":24.1,\"t\":520}]}");
+        b.append("{\"v\":22.9,\"t\":5E+2},");
+        b.append("{\"v\":22.4,\"t\":5.1E+2},");
+        b.append("{\"v\":24.1,\"t\":5.2E+2}]}");
```

Comment:
I don't like so much this way to serialize this number ?
We should maybe investigate a bit about that ?

Reply:
I tired to fix it using some suggestions from [this SO thread](https://stackoverflow.com/questions/11319445/java-to-jackson-json-serialization-money-fields), but none of them seemed to have fixed the issue. Maybe I'm doing something wrong, so you could confirm if it's actually doable.

Reply:
let's this aside for now.

Reply:
If there is no obvious/easy way to change this, I think we can live with it.

Reply:
Do you try  : https://fasterxml.github.io/jackson-core/javadoc/2.7/com/fasterxml/jackson/core/JsonGenerator.Feature.html#WRITE_BIGDECIMAL_AS_PLAIN ?

Reply:
@adamsero maybe you missed that comment ?

Reply:
I managed to finally get it working, it was a case of setting `JsonNodeFactory` to use the `withExactBigDecimals` flag.

Reply:
> I managed to finally get it working, it was a case of setting JsonNodeFactory to use the withExactBigDecimals flag.

Good to know :)


I see you change every JSON serializer, I feel this is maybe a bit out of topic.
I think it should be better to limit the change to SenML-JSON and old JSON serializer for this PR.
(I'm not even sure we want to change it for other serializer)

Reply:
Sure, I reverted the changes for all but 3 serializers